cmake/Modules/FindMKLDNN.cmake

```diff
@@ -21,10 +21,16 @@ IF(NOT MKLDNN_FOUND)
 
   if(USE_XPU) # Build oneDNN GPU library
     if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-      set(DNNL_HOST_COMPILER "g++")
+      # Linux
       # g++ is soft linked to /usr/bin/cxx, oneDNN would not treat it as an absolute path
+      set(DNNL_HOST_COMPILER "g++")
+      set(SYCL_CXX_DRIVER "icpx")
+      set(DNNL_LIB_NAME "libdnnl.a")
     else()
-      message(FATAL_ERROR "oneDNN library currently only supports GUN g++ compiler for XPU backend")
+      # Windows
+      set(DNNL_HOST_COMPILER "DEFAULT")
```

Comment:
@EikanWang @mdtoguchi @vpirogov, let's move https://github.com/pytorch/pytorch/pull/126725/files#r1618284937 discussion to here.

Reply:
> Generated binaries from icx, icpx, icx-cl on Windows will always be MSVC compatible unless there is an override via the --target option. 

Based on @mdtoguchi https://github.com/pytorch/pytorch/pull/126725#discussion_r1619027194 above, I suppose the compatibility concern of `icx` (clang under the hood) and MSVC is not an issue now with the use of `ONEDNN_DPCPP_HOST_COMPILR=DEFAULT`.

I wonder if it's the same case on Linux with `icpx` (clang under the hood) and g++. And if so, what was the original need to use `ONEDNN_DPCPP_HOST_COMPILER=g++` on Linux?

Reply:
If I recall correctly the reason to use g++ in Linux builds was threading. Pytorch relies on GNU OpenMP library and the intent was to generate GNU OpenMP code for oneDNN threading on CPU.

Reply:
> Generated binaries from icx, icpx, icx-cl on Windows will always be MSVC compatible unless there is an override via the --target option. Use of icx or icx-cl is recommended if you require MSVC compatible command line options as well. icpx will use Linux based options, but will target MSVC when run on Windows.

Sounds good. By the way, does `--target` here means that users override the `--target` and then impact the oneDNN? @mdtoguchi

Reply:
I wouldn't expect users to override the `--target` values.  The target values allow for modifying the target architecture (say x86_64 vs powerpc for example) as well as changing the OS (Windows vs Linux) or vendor (gnu vs msvc) settings.  The use of this option is useful especially in cross-compilation environments, but is not expected to be used in typical compilation scenarios.  Any usage of this option will also require the dependent libraries to be built with that same target expectation.